native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java

```diff
@@ -0,0 +1,116 @@
+        configKeys.put(SAPConstants.REPOSITORY_DESTINATION, DestinationDataProvider.JCO_REPOSITORY_DEST);
+        configKeys.put(SAPConstants.REPOSITORY_USER, DestinationDataProvider.JCO_REPOSITORY_USER);
+        configKeys.put(SAPConstants.REPOSITORY_PASSWORD, DestinationDataProvider.JCO_REPOSITORY_PASSWD);
+        configKeys.put(SAPConstants.REPOSITORY_SNC_MODE, DestinationDataProvider.JCO_REPOSITORY_SNC);
+        configKeys.put(SAPConstants.REPOSITORY_CHECK_INTERVAL, DestinationDataProvider.JCO_REPOSITORY_CHECK_INTERVAL);
+        configKeys.put(SAPConstants.TRACE, DestinationDataProvider.JCO_TRACE);
+        configKeys.put(SAPConstants.GW_HOST, DestinationDataProvider.JCO_GWHOST);
+        configKeys.put(SAPConstants.GW_SERV, DestinationDataProvider.JCO_GWSERV);
+        configKeys.put(SAPConstants.TP_HOST, DestinationDataProvider.JCO_TPHOST);
+        configKeys.put(SAPConstants.TP_NAME, DestinationDataProvider.JCO_TPNAME);
+        configKeys.put(SAPConstants.WS_HOST, DestinationDataProvider.JCO_WSHOST);
+        configKeys.put(SAPConstants.WS_PORT, DestinationDataProvider.JCO_WSPORT);
+        configKeys.put(SAPConstants.USE_TLS, DestinationDataProvider.JCO_USE_TLS);
+        configKeys.put(SAPConstants.TLS_TRUST_ALL, DestinationDataProvider.JCO_TLS_TRUST_ALL);
+        configKeys.put(SAPConstants.TLS_P12_FILE, DestinationDataProvider.JCO_TLS_P12FILE);
+        configKeys.put(SAPConstants.TLS_P12_PASSWORD, DestinationDataProvider.JCO_TLS_P12PASSWD);
+        configKeys.put(SAPConstants.TLS_CLIENT_CERTIFICATE_LOGON,
+                DestinationDataProvider.JCO_TLS_CLIENT_CERTIFICATE_LOGON);
+        configKeys.put(SAPConstants.PROXY_HOST, DestinationDataProvider.JCO_PROXY_HOST);
+        configKeys.put(SAPConstants.PROXY_PORT, DestinationDataProvider.JCO_PROXY_PORT);
+        configKeys.put(SAPConstants.PROXY_USER, DestinationDataProvider.JCO_PROXY_USER);
+        configKeys.put(SAPConstants.PROXY_PASSWORD, DestinationDataProvider.JCO_PROXY_PASSWD);
+        configKeys.put(SAPConstants.WS_PING_CHECK_INTERVAL, DestinationDataProvider.JCO_PING_CHECK_INTERVAL);
+        configKeys.put(SAPConstants.WS_PING_PERIOD, DestinationDataProvider.JCO_PING_PERIOD);
+        configKeys.put(SAPConstants.WS_PONG_TIMEOUT, DestinationDataProvider.JCO_PONG_TIMEOUT);
+        configKeys.put(SAPConstants.JCO_TYPE, DestinationDataProvider.JCO_TYPE);
+        configKeys.put(SAPConstants.USE_SAP_GUI, DestinationDataProvider.JCO_USE_SAPGUI);
+        configKeys.put(SAPConstants.DENY_INITIAL_PASSWORD, DestinationDataProvider.JCO_DENY_INITIAL_PASSWORD);
+        configKeys.put(SAPConstants.REPOSITORY_ROUNDTRIP_OPTIMIZATION,
+                DestinationDataProvider.JCO_REPOSITORY_ROUNDTRIP_OPTIMIZATION);
+
+        configKeys.forEach((key, destKey) -> {
+            String value = jcoDestinationConfig.getStringValue(StringUtils.fromString(key)).toString();
```

Comment:
Can this be null? If so should e set the property?

Reply:
Yes, It can have nulls. This is a mistake, changed with [e656056](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/2/commits/e65605635b403b106a9c0060d978049f84e6186c).